### PR TITLE
Add further actions to reduce duplication

### PR DESCRIPTION
**Pull request by amgebauer**
This PR preparing us to reuse more of the defined steps to build and test 4C in a workflow running every night.

 * Extends `build_4C` that now internally calls `configure_4C` and prepares `ccache` if wanted.
 * Adds actions to upload and download the build-folder from previous artifacts.
 * separates clang build and test in PR testing.

### CODE REVIEW

**Review by ppraegla — DISMISSED** [review]

File: .github/workflows/buildtest.yml [hunk 1]
```diff
@@ -132,27 +94,49 @@ jobs:
...
+          ./utilities/set_up_dev_env.sh
+      - uses: ./.github/actions/download_4C_build
+        with:
+          build-job: clang18_build
+      - name: Test
         run: |
           cd $GITHUB_WORKSPACE/build
           ctest -L minimal -j `nproc` --output-on-failure
+        env:
+          TEST_CHUNK: ${{ matrix.test-chunk }}
```
Comment (sebproell) [inline .github/workflows/buildtest.yml:136]: Why is this here? Does this value even exist?

Comment (amgebauer) [reply .github/workflows/buildtest.yml:136]: Ah thanks, copy paste error. I will remove it.
